### ISSUE TITLE
Type guard breaks when modifying the variable in

### ISSUE THREAD

**Issue by zoltan-mihalyi** [Microsoft/TypeScript#8857]
**TypeScript Version: 1.8.9 **

**Code**

```ts
function test(x:string|number){
    if(typeof x==='string'){
        console.log(x.length);
        x = '';
    }
}
```

**Expected behavior:**
Should not cause an error when the guarded variable is modified inside the guard scope in the following cases:

- if the variable is modified before the console.log(...) statement, and the new type is string
- if the variable is modified after the console.log(...) statement

**Actual behavior:** 
Compiling the code causes an error: Error:(3, 23) TS2339: Property 'length' does not exist on type 'string | number'.

**Comment by sandersn** [comment]
This is fixed in master. Can you try `typescript@next`?

**Comment by zoltan-mihalyi** [comment]
With `typescript@next` it works fine. Thank you!
